packages/devtools_app/lib/src/screens/inspector/inspector_service.dart

```diff
@@ -388,25 +388,26 @@ class InspectorService extends InspectorServiceBase {
   }
 
   Future<void> _updateLocalClasses() async {
-    localClasses.clear();
-    if (_rootDirectories.value.isNotEmpty) {
-      final isolate = inspectorLibrary.isolate!;
-      for (var libraryRef in isolate.libraries!) {
-        if (isLocalUri(libraryRef.uri!)) {
-          try {
-            final Library library = await inspectorLibrary.service
```

Comment:
Ah, this is probably the culprit for the worst of the slowdown. This can probably be updated to something like this:

```dart
final isolate = inspectorLibrary.isolate!;
final futures = <Future>[];
for (final libraryRef in isolate.libraries!) {
  if (isLocalUri(libraryRef.uri!)) {
    final future = inspectorLibrary.service.getObject(
        isolate.id!, libraryRef.id!
      ).then((result) {
        final library = result as Library;
        for (final classRef in library.classes!) {
          localClasses[classRef.name!] = classRef;
        }
      });
    futures.add(future);
  }
}
await Future.wait(futures);
```

This will make it so that you're not waiting on individual requests before requesting the next library, so we can really take advantage of parallelism.

Reply:
I'll add this feedback to the issue.
I'm not sure it would 100% solve the issue as the network gets flooded with requests, so it may still slow things down

Reply:
Generally speaking, multiple outstanding network requests don't slow things down nearly as much as serializing them. This is a common pattern we use when we need to make a bunch of service requests (e.g., in the [`ProgramExplorer`](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/debugger/program_explorer_controller.dart#L224) we do this to request all fields and functions in libraries and classes).

Reply:
+1 to this pattern. Can't believe I wrote taht code. Wish I had an aggressive lint that said "no await within a for loop"